src/shared/sam/cli/samCliBuild.ts

```diff
@@ -10,19 +10,71 @@ import { getLogger, Logger } from '../../logger'
 import { ChildProcessResult } from '../../utilities/childProcess'
 import { DefaultSamCliProcessInvoker, SamCliProcessInvoker } from './samCliInvoker'
 
+export interface SamCliBuildInvocationArguments {
+    /**
+     * The path to a folder where the built artifacts are stored.
+     */
+    buildDir: string,
+    /**
+     * Resolves relative paths to the function's source code with respect to this folder.
+     * If omitted, relative paths are resolved with respect to the template's location.
+     */
+    baseDir?: string,
+    /**
+     * Location of the SAM Template to build
+     */
+    templatePath: string,
+    /**
+     * Manages the sam cli execution.
+     */
+    invoker: SamCliProcessInvoker,
+    /**
+     * If your functions depend on packages that have natively compiled dependencies,
+     * use this flag to build your function inside an AWS Lambda-like Docker container.
+     */
+    useContainer?: boolean,
```

Comment:
Nit: Suggest to clarify in comments what `undefined` indicates for `userContainer` and `skipPullImage`.

Reply:
I initially did this but realized that this was an implementation/usage detail. Instead this info is in the `SamCliBuildInvocation` constructor (roughly line 58). I'm making those lines a little more verbose.